File: backend/app/application/order/commands/assign_order_item_instrument.py
"""Assign an instrument to an order item command."""

from dataclasses import dataclass
from uuid import UUID


@dataclass(frozen=True)
class AssignOrderItemInstrumentCommand:
    """Command for assigning a concrete instrument to an order item."""

    order_id: UUID
    item_id: UUID
    instrument_id: UUID

File: frontend/src/features/orders/add-order-item/api/assignOrderItemInstrument.ts
import { http } from "../../../../shared/api/http";
import { mapOrder } from "../../api/orderMapper";
import type { OrderApiDto } from "../../api/types";
import type { OrderRead } from "../../model/types";

/** Assigns a concrete instrument to an existing order item. */
export async function assignOrderItemInstrument(
  orderId: string,
  itemId: string,
  instrumentId: string,
): Promise<OrderRead> {
  const response = await http.patch<OrderApiDto>(
    `/orders/${orderId}/items/${itemId}/instrument`,
    { instrument_id: instrumentId },
  );

  return mapOrder(response.data);
}
